rfcs/0000-kuadrant-ci-cd.md

```diff
@@ -0,0 +1,837 @@
+# Kuadrant components CI/CD
+
+- Feature Name: `kuadrant-ci-cd`
+- Start Date: 2023-11-21
+- RFC PR: [Kuadrant/architecture#0038](https://github.com/Kuadrant/architecture/pull/38)
+- Issue tracking: [Kuadrant/architecture#0000](https://github.com/Kuadrant/architecture/issues/0000)
+
+# Summary
+[summary]: #summary
+
+This RFC focuses on how to build and deliver Kuadrant components, in a new and more agile way of getting
+the desired "artifacts" (images, manifests, etc.) into the hands of the users, devs, QE team and any other process that
+needs them.
+
+# Motivation
+[motivation]: #motivation
+
+The current process is not only slow, but also involves manual steps that are prone to human error. Its implementation
+is also not very flexible, and it's not easy to add new steps or change the existing ones without replicating the same
+code in different repositories. It also involves a convolution of different tools and services that are not easy to
+maintain and that are not very well integrated with each other.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+## Current process
+
+<details>
+<summary>Summary of the current process</summary>
```

Comment:
I am missing DNS operator and MGC for both current and new process

Reply:
I added (not yet merged) a release doc to the dns operator repo https://github.com/Kuadrant/dns-operator/blob/88185359c731d41d8909c1ae45bf0a6ec14d0031/docs/RELEASE.md

Similar to what Gui describes here https://github.com/Kuadrant/architecture/pull/41/files#r1444455110, it has a file generated by a `prepare-release` tsk that is saved to the release branch, but its just a [Makefile](https://github.com/Kuadrant/dns-operator/blob/release-0.1/make/release.mk) overriding the appropriate variables so that all the make targets function as expected if you were to run them on that branch.

Do we want all this copied into this doc, or can we just link to that file?

re. MGC, not sure what we did for a release there tbh, but it's an operator like most others and we probably just manually did it all. Branching tagging would be the same as what is described in the dns-operator doc.

Reply:
Yes, that should be added here too, since we're gonna be using this document's section to gather every detail of the current release process of the components. Thanks!